src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerTest.java

```diff
@@ -1888,6 +1889,128 @@ public void testDynamicTriggerConfigurationTimeGap() throws Exception {
+    }
+
+    /**
+     * Tests that dynamic project configurations failing with exception does not block the event listener.
+     * @throws Exception on failure
+     */
+    @PrepareForTest({
+            GerritTrigger.class,
+            AbstractProject.class,
+            ToGerritRunListener.class,
+            PluginImpl.class,
+            Hudson.class,
+            Jenkins.class,
+            DependencyQueueTaskDispatcher.class,
+            DynamicConfigurationCacheProxy.class,
+            EventListener.class })
+    @Test
+    public void testDynamicTriggerUpdateFailureDoesNotBlock() throws Exception {
+        AbstractProject project = PowerMockito.mock(AbstractProject.class);
+        when(project.getFullName()).thenReturn("MockedProject");
+        when(project.isBuildable()).thenReturn(true);
+
+        DynamicConfigurationCacheProxy configurationCacheProxy = PowerMockito.mock(DynamicConfigurationCacheProxy.class);
+        PowerMockito.mockStatic(DynamicConfigurationCacheProxy.class);
+        when(DynamicConfigurationCacheProxy.getInstance()).thenReturn(configurationCacheProxy);
+        when(configurationCacheProxy.fetchThroughCache("url")).thenThrow(new RuntimeException());
+
+        Queue queue = mockConfig(project);
+
+        PowerMockito.mockStatic(ToGerritRunListener.class);
+        ToGerritRunListener listener = PowerMockito.mock(ToGerritRunListener.class);
+        PowerMockito.when(ToGerritRunListener.getInstance()).thenReturn(listener);
+
+        GerritTrigger trigger = new GerritTrigger(null);
+        trigger.setDynamicTriggerConfiguration(true);
+        trigger.setTriggerConfigURL("url");
+
+        // Set up the job within the trigger.
+        Whitebox.setInternalState(trigger, "job", project);
+
+        // We need to make sure that whenever anyone asks for the trigger for any job
+        // that it returns this one.
+        PowerMockito.mockStatic(GerritTrigger.class);
+        when(GerritTrigger.getTrigger(any(Job.class))).thenReturn(trigger);
+        assertEquals(trigger, GerritTrigger.getTrigger(project));
+
+        // Because the "stub" methodology doesn't work, we also need to manually replace the "createListener"
+        // static method.
+        EventListener myListener = new EventListener(project);
+        PowerMockito.when(GerritTrigger.createListener(any(Job.class))).thenReturn(myListener);
+        assertNotNull(trigger.createListener());
+
+        // Start the trigger.
+        trigger.start(project, true);
+
+        // Make sure that the timer task started.
+        assertNotNull(Whitebox.getInternalState(trigger, "gerritTriggerTimerTask"));
+
+        // Wait until the timer task has run for the first time.
+        Thread.sleep(GerritTriggerTimer.DELAY_MILLISECONDS + 1000);
+
+        CountDownLatch latch = Whitebox.getInternalState(trigger, "projectListIsReady");
+        assertNotNull(latch);
+        assertEquals(0, latch.getCount());
+    }
+
```

Comment:
You should also try to make a non mocked integration test (just the happy path), my guess it that it would be less complex to write than this version with mocks ;)

Reply:
I can try, checking the blocking condition was more important to me and I just assumed the happy case is already covered.